apps/cowswap-frontend/src/common/updaters/orders/utils.ts

```diff
@@ -19,55 +20,53 @@ export function computeOrderSummary({
   orderFromStore?: Order
   orderFromApi: EnrichedOrder | null
 }) {
-  // Default to store's current order summary
-  let summary: string | undefined = orderFromStore?.summary
-
-  // if we can find the order from the API
-  // and our specific order exists in our state, let's use that
-  if (orderFromApi) {
-    const {
-      buyToken,
-      sellToken,
-      sellAmount,
-      feeAmount,
-      buyAmount,
-      executedBuyAmount,
-      executedSellAmount,
-      owner,
-      receiver,
-    } = orderFromApi
-
-    if (orderFromStore) {
-      const { inputToken, outputToken, status, kind } = orderFromStore
-      const isFulfilled = status === OrderStatus.FULFILLED
-
-      if (!inputToken || !outputToken) return undefined
-
-      // don't show amounts in atoms
-      const inputAmount = isFulfilled
-        ? stringToCurrency(executedSellAmount, inputToken)
-        : // sellAmount doesn't include the fee, so we add it back to not show a different value when the order is traded
-          stringToCurrency(sellAmount, inputToken).add(stringToCurrency(feeAmount, inputToken))
-      const outputAmount = stringToCurrency(isFulfilled ? executedBuyAmount : buyAmount, outputToken)
-
-      const isSell = isSellOrder(kind)
-
-      const inputPrefix = !isFulfilled && !isSell ? 'at most ' : ''
-      const outputPrefix = !isFulfilled && isSell ? 'at least ' : ''
-
-      summary = `Swap ${inputPrefix}${formatTokenAmount(inputAmount)} ${formatSymbol(
-        inputAmount.currency.symbol
-      )} for ${outputPrefix}${formatTokenAmount(outputAmount)} ${formatSymbol(outputAmount.currency.symbol)}`
-    } else {
-      // We only have the API order info, let's at least use that
-      summary = `Swap ${sellToken} for ${buyToken}`
-    }
-
-    if (owner && receiver && receiver !== owner) {
-      summary += ` to ${shortenAddress(receiver)}`
-    }
+  if (!orderFromStore && !orderFromApi) return undefined
+
+  const buyToken = orderFromApi?.buyToken || orderFromStore?.buyToken
```

Comment:
In the code above we explicitly exit from function when both orders are falsy, it means that here at least one of them is truthy.
But typescript is not so smart, so I had to cast to string some consts.